scheduler/core/eventsqueue/events.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List
+
+from lucupy.minimodel import Resource, Conditions
+
+from scheduler.core.meta import AbstractDataclass, FrozenAbstractDataclass
+
+
+@dataclass(frozen=True)
+class Interruption(FrozenAbstractDataclass):
+    """
+    Parent class for any interruption in the night that would
+    cause a new schedule to be created.
+    """
```

Comment:
This probably should be:
```
Parent class for any interruption that could cause the generation of a new plan.
```

The idea is for these events to be checked and determine if we should create a new plan based on the information provided.